bdk_core/tests/test_sparse_chain.rs

```diff
@@ -235,7 +229,7 @@ fn adding_checkpoint_which_contains_nothing_new_doesnt_create_new_checkpoint() {
         ..Default::default()
     };
     assert_eq!(chain.apply_checkpoint(update.clone()), ApplyResult::Ok);
-    assert_eq!(chain.iter_checkpoints(..).count(), 1);
+    assert_eq!(chain.iter_checkpoints(..).count(), 2);
```

Comment:
we need to fix the name of the test since it should create a new checkpoint!

Perhaps we just delete the test for now unless we can think of a better way to test it.